graphrag/index/flows/create_final_community_reports.py

```diff
@@ -0,0 +1,167 @@
+# Copyright (c) 2024 Microsoft Corporation.
+# Licensed under the MIT License
+
+"""All the steps to transform community reports."""
+
+from uuid import uuid4
+
+import pandas as pd
+from datashaper import (
+    AsyncType,
+    VerbCallbacks,
+)
+
+from graphrag.index.cache import PipelineCache
+from graphrag.index.graph.extractors.community_reports.schemas import (
+    CLAIM_DESCRIPTION,
+    CLAIM_DETAILS,
+    CLAIM_ID,
+    CLAIM_STATUS,
+    CLAIM_SUBJECT,
+    CLAIM_TYPE,
+    EDGE_DEGREE,
+    EDGE_DESCRIPTION,
+    EDGE_DETAILS,
+    EDGE_ID,
+    EDGE_SOURCE,
+    EDGE_TARGET,
+    NODE_DEGREE,
+    NODE_DESCRIPTION,
+    NODE_DETAILS,
+    NODE_ID,
+    NODE_NAME,
+)
+from graphrag.index.verbs.graph.report.create_community_reports import (
+    create_community_reports_df,
+)
+from graphrag.index.verbs.graph.report.prepare_community_reports import (
+    prepare_community_reports_df,
+)
+from graphrag.index.verbs.graph.report.restore_community_hierarchy import (
+    restore_community_hierarchy_df,
+)
+from graphrag.index.verbs.text.embed.text_embed import text_embed_df
+
+
+async def create_final_community_reports(
+    nodes_input: pd.DataFrame,
+    edges_input: pd.DataFrame,
+    claims_input: pd.DataFrame | None,
+    callbacks: VerbCallbacks,
+    cache: PipelineCache,
+    strategy: dict,
+    async_mode: AsyncType = AsyncType.AsyncIO,
+    num_threads: int = 4,
+    covariates_enabled: bool = False,
+    full_content_text_embed: dict | None = None,
+    summary_text_embed: dict | None = None,
+    title_text_embed: dict | None = None,
+) -> pd.DataFrame:
+    """All the steps to transform community reports."""
+    nodes = _prep_nodes(nodes_input)
+    edges = _prep_edges(edges_input)
+
+    claims = None
+    if covariates_enabled and claims_input is not None:
+        claims = _prep_claims(claims_input)
+
+    community_hierarchy = restore_community_hierarchy_df(nodes)
+
+    local_contexts = prepare_community_reports_df(
+        nodes, edges, claims, callbacks, strategy.get("max_input_length", 16_000)
+    )
+
+    community_reports = await create_community_reports_df(
+        local_contexts,
+        nodes,
+        community_hierarchy,
+        callbacks,
+        cache,
+        strategy,
+        async_mode=async_mode,
+        num_threads=num_threads,
+    )
+
+    community_reports["id"] = community_reports["community"].apply(
+        lambda _x: str(uuid4())
```

Comment:
Wonder if we could replace this apply with someting like:
`community_reports["id"] = pd.util.hash_pandas_object(community_reports.index).astype(str)`

These ids are just random, right?

Reply:
I do not know what uniqueness guarantees pandas makes with that function versus uuid